### ISSUE TITLE
How do i create a new model from my epoch.pth checkpoint file and config file?

### ISSUE THREAD

**Issue by Enish258** [open-mmlab/mmdetection#5354]
I have a checkpoint file that I created by train_detector.
![Screenshot (449)](https://user-images.githubusercontent.com/64497088/121950960-ef07cc00-cd77-11eb-8c06-f6d8bf1d4e79.png).
After that, let's say I have to create a new model.So what exactly is the procedure for that?
Like should I create an init_detector using my saved epoch checkpoint file +the original model cfg?
I did it this way.
![Screenshot (451)](https://user-images.githubusercontent.com/64497088/121951940-35a9f600-cd79-11eb-8911-a1391c5ce2f9.png)

![Screenshot (450)](https://user-images.githubusercontent.com/64497088/121951671-d77d1300-cd78-11eb-9f66-52b529fa6c63.png)
But it keeps giving a warning saying that class names are not saved in checkpoints.And when i do inference using the model,it keeps outputting the original classes itself instead of the trained classes.
![Screenshot (452)](https://user-images.githubusercontent.com/64497088/121952157-815c9f80-cd79-11eb-9774-50743b4bd50a.png)
Please help me.
Thank you.

**Comment by hhaAndroid** [comment]
Hi @Enish258 Why don't you use the https://github.com/open-mmlab/mmdetection/blob/master/demo/image_demo.py script directly for inference.

**Comment by Enish258** [comment]
Okay.But actually the confusion I have is ,how do i use inference detector for testing using my own checkpoint.Like why does it not get the correct classes.Any kind of demo or anything.Pls,if you can tell me that procedure it would be extremely helpful.

**Comment by Enish258** [comment]
I actually am kinda new to mmdetection

**Comment by Enish258** [comment]
In the link you gave also I think I would need to pass config file and checkpoint file.Now I used finetuning on a pretrained VFNet model.That checkpoint is saved as epoch3.pth.Now ,I would pass this as a checkpoint file right.Now do i pass config file directly(the VFNet file) or do i make some changes .Like do I have to do things like cgf.classes=14 and all that even before running the file you mentioned or can i run the config file directly?

**Comment by hhaAndroid** [comment]
> Okay.But actually the confusion I have is ,how do i use inference detector for testing using my own checkpoint.Like why does it not get the correct classes.Any kind of demo or anything.Pls,if you can tell me that procedure it would be extremely helpful.

If you directly use the mmdet framework for training, then the class information will be automatically maintained.

**Comment by Enish258** [comment]
I did use the mmdet framework.
![Screenshot (453)](https://user-images.githubusercontent.com/64497088/122022084-e4831c00-cde3-11eb-82e5-5d4d25f998a8.png)
What I am asking is do i need to make any changes in the already available VFNet config file during testing.Do I use the already available VFNet config file [https://github.com/open-mmlab/mmdetection/blob/master/configs/vfnet/vfnet_r50_fpn_mdconv_c3-c5_mstrain_2x_coco.py] while passing config arg?Coz in that the num_classes=80 whereas for me num_classes=14 and so on.
![Screenshot (454)](https://user-images.githubusercontent.com/64497088/122022810-96bae380-cde4-11eb-8265-0ff188856121.png)
And what do i pass in the checkpoint file-can I pass epoch_3.pth,the trained checkpoint file directly or do i make certain changes in that too?

**Comment by Enish258** [comment]
And one more question, when I do training using train_detector,is a new config file created in work_dir or do we use the original config file each time?

**Comment by hhaAndroid** [comment]
> And one more question, when I do training using train_detector,is a new config file created in work_dir or do we use the original config file each time?

This configuration is just for backup

**Comment by SourabhRanade** [comment]
@hhaAndroid @Enish258 how was the issue solved exactly? I am sorry but I do not see the exact solution in the thread. I am facing a similar issue right now with a MASK RCNN using SWIN transformer as a background using MMDetection which I am trying to modify 
Could you please direct me to the solution/approach used?

**Comment by PDahal2871** [comment]
How was this issue solved? I have also a saved .pth file. I want it for inference, but still no luck.
